src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -309,12 +309,101 @@ void CoreWorkerDirectTaskReceiver::Init(
   client_pool_ = client_pool;
 }
 
-void CoreWorkerDirectTaskReceiver::HandlePushTask(
+void CoreWorkerDirectTaskReceiver::HandleActorTask(
     const rpc::PushTaskRequest &request, rpc::PushTaskReply *reply,
     rpc::SendReplyCallback send_reply_callback) {
   RAY_CHECK(waiter_ != nullptr) << "Must call init() prior to use";
   const TaskSpecification task_spec(request.task_spec());
 
+  // Make sure this is an actor task. ActorCreationTask(s) are normal tasks.
+  RAY_CHECK(task_spec.IsActorTask() && !task_spec.IsActorCreationTask());
+
+  // Only assign resources for non-actor tasks. Actor tasks inherit the resources
+  // assigned at initial actor creation time.
+  std::shared_ptr<ResourceMappingType> resource_ids;
```

Comment:
It's hard to tell from the code diff, but is most of this code the same as for `EnqueueNormalTask`? Could we pull out the common code into a separate method?